packages/core/src/implementations/business/QueryService.ts

```diff
@@ -757,4 +812,98 @@ export class QueryService implements IQueryService {
     }
     return okAsync(undefined);
   }
+
+  protected parseDynamicRewardParameter(
+    rewardParams: JSONString,
+  ): boolean | IDynamicRewardParameter[] {
```

Comment:
Why this return type? That's super confusing. If you need to represent that the dynamic rewards don't parse, return null, or better yet, use a Result and return an error state.